### PR TITLE
7TV control panel

### PR DESCRIPTION
**Pull request by Excellify**
Improves the 7tv settings menu to be more of a controll center

Right now only the settings part is implemented

### CODE REVIEW

File: src/site/twitch.tv/modules/settings/areas/SettingsArea.vue [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+const props = defineProps<{
+	category: string;
+	subcategory: string;
+	keys?: Record<string, string[]>;
+}>();
+
+const refs = ref([] as HTMLElement[]);
+
+const nodes = computed(getNodes);
+const settings = computed(() => {
```
Comment (AnatoleAM) [inline src/site/twitch.tv/modules/settings/areas/SettingsArea.vue:31]: Don't do complex `computed` calls. Also don't return a `Map`. Define a ref instead

File: src/site/twitch.tv/modules/settings/SettingsMenuButton.vue [hunk 2]
```diff
@@ -20,20 +20,50 @@ const emit = defineEmits<{
...
+if (!menuButtons) {
+	// Sidenav in modview
+	menuButtons = document.querySelector(".modview-dock__followed-channels")?.nextSibling;
+	tooltip = "right";
+}
+if (!menuButtons) {
+	// Header if popout
+	menuButtons = document.querySelector(".stream-chat-header")?.lastChild;
+	tooltip = "left";
+}
```
Comment (AnatoleAM) [inline src/site/twitch.tv/modules/settings/SettingsMenuButton.vue:38]: This is extremely KKona

File: src/site/twitch.tv/modules/settings/SettingsMenu.vue [hunk 3]
```diff
@@ -1,78 +1,146 @@
...
+			nodes[category][subcategory].push(key);
+		}
+	}
+
+	return new Map(
+		Object.entries(nodes).sort(([a], [b]) => {
+			return (t[a]._order as number) - (t[b]._order as number);
+		}),
+	);
+});
```
Comment (AnatoleAM) [inline src/site/twitch.tv/modules/settings/SettingsMenu.vue:122]: Don't do complex `computed` calls. Also don't return a `Map`. Define a ref instead

Comment (Excellify) [reply src/site/twitch.tv/modules/settings/SettingsMenu.vue:122]: I tested, and its only called once when opening the settings and when searching. Whats wrong with returning a map?

Comment (AnatoleAM) [reply src/site/twitch.tv/modules/settings/SettingsMenu.vue:122]: This is brokeback though, just watch the data you need and assign it to a ref declared at the top level. You don't need to do this two-step process, it's unreadable and inefficient

File: src/site/twitch.tv/modules/settings/SettingsNode.vue [hunk 4]
```diff
@@ -1,14 +1,16 @@
 <template>
-	<div class="settings-node">
-		<div class="label">
-			{{ node.label }}
-		</div>
-		<div class="component-container">
-			<component :is="getComponent(node)" :node="node" />
-			<div v-if="node.hint" class="hint">
-				{{ node.hint }}
+	<div class="settings-node" tabindex="0" :title="node.hint">
```
Comment (AnatoleAM) [inline src/site/twitch.tv/modules/settings/SettingsNode.vue:2]: The usage of the `title` attribute here is slightly obnoxious. It's also redundant as it's already written in pretty much the same place the tooltip appear.

File: src/site/twitch.tv/modules/settings/areas/SettingsArea.vue [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+		}
+		temp[sub].sort((a, b) => a.label.localeCompare(b.label));
+	}
+	return Object.entries(temp).sort(([a], [b]) => (a == "" ? 1 : a.localeCompare(b)));
+});
+
+onUpdated(() => {
+	const target = refs.value.find((r) => r.id == props.subcategory);
+	target && target.scrollIntoView();
+});
```
Comment (AnatoleAM) [inline src/site/twitch.tv/modules/settings/areas/SettingsArea.vue:47]: Confused, why are we doing this here? Please find a better way to achieve this

File: src/site/twitch.tv/modules/settings/SettingsNode.vue (hunk 4 above)
Comment (Excellify) [reply src/site/twitch.tv/modules/settings/SettingsNode.vue:2]: Not if the setting menu is in vertical mode

**Pull request merged**